### PR TITLE
BugFix: Serialize max_retry_delay as a timedelta

### LINKED ISSUE #13086: max_retry_delay should be a timedelta for type hinting
**Apache Airflow version**:

master

**What happened**:

https://github.com/apache/airflow/blob/master/airflow/models/baseoperator.py#L356 --> should be timedelta not datetime

### PR DESCRIPTION
**Pull request by paulvic**
closes: #13086, #14212

Whilst testing upgrading from 1.10.12 to 2.0.1 I've been seeing the following scheduler error:
```
Traceback (most recent call last):
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/jobs/scheduler_job.py", line 1280, in _execute
    self._run_scheduler_loop()
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/jobs/scheduler_job.py", line 1382, in _run_scheduler_loop
    num_queued_tis = self._do_scheduling(session)
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/jobs/scheduler_job.py", line 1521, in _do_scheduling
    self._schedule_dag_run(dag_run, active_runs_by_dag_id.get(dag_run.dag_id, set()), session)
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/jobs/scheduler_job.py", line 1760, in _schedule_dag_run
    schedulable_tis, callback_to_run = dag_run.update_state(session=session, execute_callbacks=False)
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/utils/session.py", line 62, in wrapper
    return func(*args, **kwargs)
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/models/dagrun.py", line 406, in update_state
    info = self.task_instance_scheduling_decisions(session)
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/utils/session.py", line 62, in wrapper
    return func(*args, **kwargs)
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/models/dagrun.py", line 507, in task_instance_scheduling_decisions
    schedulable_tis, changed_tis = self._get_ready_tis(scheduleable_tasks, finished_tasks, session)
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/models/dagrun.py", line 535, in _get_ready_tis
    session=session,
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/utils/session.py", line 62, in wrapper
    return func(*args, **kwargs)
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 838, in are_dependencies_met
    for dep_status in self.get_failed_dep_statuses(dep_context=dep_context, session=session):
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 859, in get_failed_dep_statuses
    for dep_status in dep.get_dep_statuses(self, session, dep_context):
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/ti_deps/deps/base_ti_dep.py", line 101, in get_dep_statuses
    yield from self._get_dep_statuses(ti, session, dep_context)
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/ti_deps/deps/not_in_retry_period_dep.py", line 47, in _get_dep_statuses
    next_task_retry_date = ti.next_retry_datetime()
  File "/app/.heroku/python/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 905, in next_retry_datetime
    delay = min(self.task.max_retry_delay, delay)
TypeError: '<' not supported between instances of 'datetime.timedelta' and 'float'
```

It appears that the `max_retry_delay` value is not being serialised as a timedelta type. This PR seeks to address that.

### CODE REVIEW

File: tests/serialization/test_dag_serialization.py [hunk 1]
```diff
@@ -856,7 +860,7 @@ def test_no_new_fields_added_to_base_operator(self):
             'executor_config': {},
             'inlets': [],
             'label': '10',
-            'max_retry_delay': None,
+            'max_retry_delay': timedelta(0, 600),
```
Comment (kaxil) [inline tests/serialization/test_dag_serialization.py]: This should still be None as we haven't changed the default value in the BaseOperator constructor

Comment (paulvic) [reply tests/serialization/test_dag_serialization.py]: Gotcha 👍

**Comment by paulvic** [comment]
Hi @kaxil, it appears that the `build docs` check was cancelled, but I don't see any way to re-trigger it. Any advice on how to proceed?

**Review by kaxil — APPROVED** [review]

**Comment by paulvic** [comment]
thanks @kaxil 🙇

**Pull request merged**